server/server.go

```diff
@@ -355,6 +366,9 @@ func (bs *Server) sendBlocks(ctx context.Context, env *decision.Envelope) {
 	bs.counters.DataSent += uint64(dataSent)
 	bs.counterLk.Unlock()
 	bs.sentHistogram.Observe(float64(env.Message.Size()))
+	if bs.onDataSentListener != nil {
+		bs.onDataSentListener(env.Peer, uint64(dataSent), uint64(len(blocks)))
```

Comment:
Call this synchronously ?

Reply:
do you mean asynchronously?

Reply:
I would have expected this to be `go bs.onDataSentListener(env.Peer, uint64(dataSent), uint64(len(blocks)))`, just wanted to be it's fine to do a blocking call here.